### ISSUE TITLE
S3052 FP: Do not raise for default values in field structs for C# versions < 11.

### ISSUE THREAD

**Issue by mary-georgiou-sonarsource** [SonarSource/sonar-dotnet#6461]
### Description

For C# versions before 11 if there is a field declared in a struct, it has to be initialized otherwise there is a compiler error ([CS0171](https://learn.microsoft.com/en-us/dotnet/csharp/misc/cs0171?f1url=%3FappId%3Droslyn%26k%3Dk(CS0171))).

The rule should not raise an issue in this case, as there is no other option for the user. 

### Repro steps

```csharp
  public struct TestStruct
  {
      public int someField = 0; //  FP for versions < C# 11 there's no default values for fields and the user needs to set it explicitly.
      public TestStruct()
      { }
  }
```

In C# 11 [auto-default structs](https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#auto-default-struct) were introduced, so from this version version onwards the compiler ensures that all fields of a struct type are initialized to their default value.

### Expected behavior

No issue should be raised for structs for language versions < C# 11

### Related information

* C#/VB.NET Plugins version: 8.49

**Comment by martin-strecker-sonarsource** [comment]
Note: struct field initializers are a C#9 feature. The rule should not trigger between versions 9 and 10 but from version 11 onwards it should.

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1797](https://sonarsource.atlassian.net/browse/NET-1797)

[NET-1797]: https://sonarsource.atlassian.net/browse/NET-1797?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
